`make dist` failing

**Environment**
- libunwind version:
    current master branch. Also observed on 1.8.1.
- architecture:
    aarch64
- OS:
    Linux Ubuntu 22.04.1 aarch64

**Description**
The "make dist" target is meant to produce a distributable source tarball.
I believe this used to work at some point. I'm not sure how to debug it, sorry.
I have no sense of how autoconf is supposed to be set up.

**Reproduction Steps**

Fresh checkout. configure and `make dist`.
```sh
git clone git@github.com:libunwind/libunwind
pushd libunwind
autoreconf -i
./configure
make dist
```

Result:

```
 make dist
make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/tmp/unwind_make_dist/libunwind'
make  distdir-am
make[2]: Entering directory '/tmp/unwind_make_dist/libunwind'
if test -d "libunwind-1.9-pre"; then find "libunwind-1.9-pre" -type d ! -perm -200 -exec chmod u+w {} ';' && rm -rf "libunwind-1.9-pre" || { sleep 5 && rm -rf "libunwind-1.9-pre"; }; else :; fi
test -d "libunwind-1.9-pre" || mkdir "libunwind-1.9-pre"
 (cd src && make  top_distdir=../libunwind-1.9-pre distdir=../libunwind-1.9-pre/src \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory '/tmp/unwind_make_dist/libunwind/src'
make  distdir-am
make[4]: Entering directory '/tmp/unwind_make_dist/libunwind/src'
make[4]: Leaving directory '/tmp/unwind_make_dist/libunwind/src'
make[3]: Leaving directory '/tmp/unwind_make_dist/libunwind/src'
 (cd tests && make  top_distdir=../libunwind-1.9-pre distdir=../libunwind-1.9-pre/tests \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory '/tmp/unwind_make_dist/libunwind/tests'
make  distdir-am
make[4]: Entering directory '/tmp/unwind_make_dist/libunwind/tests'
make[4]: Leaving directory '/tmp/unwind_make_dist/libunwind/tests'
make[3]: Leaving directory '/tmp/unwind_make_dist/libunwind/tests'
 (cd doc && make  top_distdir=../libunwind-1.9-pre distdir=../libunwind-1.9-pre/doc \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory '/tmp/unwind_make_dist/libunwind/doc'
make[3]: *** No rule to make target 'distdir'.  Stop.
make[3]: Leaving directory '/tmp/unwind_make_dist/libunwind/doc'
make[2]: *** [Makefile:729: distdir-am] Error 1
make[2]: Leaving directory '/tmp/unwind_make_dist/libunwind'
make[1]: *** [Makefile:723: distdir] Error 2
make[1]: Leaving directory '/tmp/unwind_make_dist/libunwind'
make: *** [Makefile:829: dist] Error 2
```

It seemed to run into an error in the `doc/` directory.

Just a quick follow-up on this.

We're using libunwind code as part of a large project and we aren't bringing in its docs and tests.
So we `./configure --disable-documentation --disable-tests` and some other disables.

So after installing latex2man, I can now do a basic `./configure` / `make dist` sequence.
But this fails if I give those disable options to configure in a way similar to the initial report here.

```
make[3]: Entering directory '/tmp/import-libunwind.QOCvBn/repo/tests'
make[3]: *** No rule to make target 'distdir'.  Stop.
```

or 

```
make[3]: Entering directory '/tmp/import-libunwind.chdShd/repo/doc'
make[3]: *** No rule to make target 'distdir'.  Stop.
```

It seems like these disables aren't working properly.

--- a/configure.ac
+++ b/configure.ac
@@ -317,18 +317,22 @@ AM_CONDITIONAL([SUPPORT_CXX_EXCEPTIONS], [test x$enable_cxx_exceptions = xyes])
 
 AC_MSG_CHECKING([if documentation should be built])
 AC_ARG_ENABLE([documentation],
-	          [AS_HELP_STRING([--enable-documentation],
-	                          [enable generating the man pages @<:@default=yes@:>@])],
-	          [],
-	          [enable_documentation=yes])
+              [AS_HELP_STRING([--enable-documentation],
+                              [enable generating the man pages @<:@default=yes@:>@])],
+              [],
+              [enable_documentation=yes])
 AC_MSG_RESULT([$enable_documentation])
-AC_PATH_PROG([LATEX2MAN],[latex2man])
-AS_IF([test "x$LATEX2MAN" = "x" && test "x$enable_documentation" != "xno"], [
-  AC_MSG_WARN([latex2man not found. Install latex2man. Disabling docs.])
-  enable_documentation="no";
-])
 AM_CONDITIONAL([CONFIG_DOCS], [test x$enable_documentation != xno])
-AM_COND_IF([CONFIG_DOCS], [AC_CONFIG_FILES([doc/Makefile doc/common.tex])])
+AM_COND_IF([CONFIG_DOCS],
+           [AC_CONFIG_FILES([doc/Makefile doc/common.tex])
+            AC_PATH_PROG([LATEX2MAN],[latex2man],[:])
+            AS_IF([test "x$LATEX2MAN" = "x:"],
+                  [AC_MSG_WARN([latex2man not found. Can not build man pages.])])
+            AC_PATH_PROG([PDFLATEX],[pdflatex],[:])
+            AS_IF([test "x$PDFLATEX" = "x:"],
+                  [AC_MSG_WARN([pdflatex not found. Can not build PDF documentation.])])
+           ]
+)
 
 # Enable tests built around unw_resume, which is not supported on all targets
 AC_MSG_CHECKING([if we should enable unw_resume tests])

--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -72,8 +72,8 @@ EXTRA_DIST = NOTES libunwind.trans					\
 	unw_get_elf_filename_by_ip.tex	\
 	$(man3_MANS)
 
-L2M	= latex2man
-L2P	= pdflatex
+L2M	= ${LATEX2MAN}
+L2P	= ${PDFLATEX}
 L2M_CMD	= $(L2M) -t $(srcdir)/libunwind.trans
 L2H_CMD	= $(L2M) -H -t $(srcdir)/libunwind.trans
 

